tests/unit_tests/test_config.py

```
@@ -92,3 +92,41 @@ def test_override_mps(monkeypatch, tiny_config, tmp_path, caplog):
+    """Test that a non-mapping YAML file raises TypeError."""
+    filename = str(tmp_path / "config_list.yml")
+    with open(filename, "w") as f:
+        f.write("- item1\n- item2\n")
+    with pytest.raises(TypeError, match="must contain a YAML mapping"):
+        Config(filename)
+
+
+def test_deprecated_only_old_key(tmp_path, tiny_config):
+    """Test deprecated key is remapped when only the old key is present."""
+    filename = str(tmp_path / "config_deprecated_remap.yml")
+    with open(tiny_config, "r") as f_in, open(filename, "w") as f_out:
```

Ruff (0.15.6)

[warning] Unnecessary mode argument

Remove mode argument

(UP015)